azurerm/resource_arm_mariadb_server.go

```diff
@@ -0,0 +1,381 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/mariadb/mgmt/2018-06-01-preview/mariadb"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMariaDbServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMariaDbServerCreateOrUpdate,
+		Read:   resourceArmMariaDbServerRead,
+		Update: resourceArmMariaDbServerCreateOrUpdate,
+		Delete: resourceArmMariaDbServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"sku": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"B_Gen5_1",
+								"B_Gen5_2",
+								"GP_Gen5_2",
+								"GP_Gen5_4",
+								"GP_Gen5_8",
+								"GP_Gen5_16",
+								"GP_Gen5_32",
+								"MO_Gen5_2",
+								"MO_Gen5_4",
+								"MO_Gen5_8",
+								"MO_Gen5_16",
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"capacity": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ValidateFunc: validate.IntInSlice([]int{
+								1,
+								2,
+								4,
+								8,
+								16,
+								32,
+							}),
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(mariadb.Basic),
+								string(mariadb.GeneralPurpose),
+								string(mariadb.MemoryOptimized),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
(same here) can we make this case sensitive?

Reply:
Done.